src/qtt/measurements/new/uhfli_scope_reader.py

```diff
@@ -0,0 +1,242 @@
+from typing import List
+import numpy as np
+import time
+
+from qilib.configuration_helper import InstrumentAdapterFactory
+from qilib.data_set import DataArray, DataSet
+from qilib.utils import PythonJsonStructure
+from qtt.measurements.new.interfaces import AcquisitionScopeInterface
+from qcodes.instrument_drivers.ZI import ZIUHFLI
+
+
+class UhfliScopeReader(AcquisitionScopeInterface):
+    """ Represents an acquisition wrapper for acquiring data with the Zurich Instruments UHFLI."""
+
+    def __init__(self, address):
+        """ Creates the connection to the UHFLI lock-in amplifier.
+
+        Args:
+            address: The unique address of the UHFLI.
+        """
+        self.adapter = InstrumentAdapterFactory.get_instrument_adapter('ZIUHFLIInstrumentAdapter', address)
+        self.__uhfli = self.adapter.instrument
+        self.__acquisition_counter = 0
+        self.__address = address
+
+    def initialize(self, configuration: PythonJsonStructure) -> None:
+        """ Applies the configuration to the UHFLI.
+
+        Args:
+            config: The configuration with all the UHFLI settings.
+        """
+        self.adapter.apply(configuration)
+
+    def prepare_acquisition(self) -> None:
+        """ Sets the UFHLI into scope mode such that the device can collect traces."""
+        self.__uhfli.scope.set('scopeModule/mode', 1)
+        self.__uhfli.scope.subscribe('/{0}/scopes/0/wave'.format(self.__address))
+        self.__uhfli.daq.sync()
+        self.__uhfli.daq.setInt('/{0}/scopes/0/enable'.format(self.__address), 1)
```

Comment:
I think not, because we call acquire multiple times. We do not want to enable the acquisition every time we collect data. That makes the readout slow.